packages/astro/src/prefetch/index.ts

```diff
@@ -215,6 +215,11 @@ export interface PrefetchOptions {
  * @param opts Additional options for prefetching.
  */
 export function prefetch(url: string, opts?: PrefetchOptions) {
+	// Remove url hash to avoid prefetching the same URL multiple times
+	const urlObj = new URL(url, location.href);
+	urlObj.hash = '';
+	url = urlObj.toString();
```

Comment:
```suggestion
	// Remove url hash to avoid prefetching the same URL multiple times
	url = url.replace(/#.*/, '');
```

I think we can simplify it as this instead so we can skip the overhead of `new URL` and `.toString`